cmd/ingress-operator/start.go

```diff
@@ -65,76 +73,66 @@ func NewStartCommand() *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringVarP(&options.ShutdownFile, "shutdown-file", "s", defaultTrustedCABundle, "file which when modified should trigger a shutdown. If empty, shutdown triggering is disabled.")
+	cmd.Flags().StringVarP(&options.ShutdownFile, "shutdown-file", "s", defaultTrustedCABundle, "if provided, shut down the operator when this file changes")
+	cmd.Flags().StringVarP(&options.MetricsListenAddr, "metrics-listen-addr", "", ":6000", "metrics endpoint listen address")
+	cmd.Flags().StringVarP(&options.OperatorNamespace, "namespace", "n", manifests.DefaultOperatorNamespace, "namespace the operator is deployed to")
+	cmd.Flags().StringVarP(&options.IngressControllerImage, "image", "i", "", "image of the ingress controller the operator will manage (required)")
+	cmd.Flags().StringVarP(&options.ReleaseVersion, "release-version", "", statuscontroller.UnknownVersionValue, "the release version the operator should converge to")
+
+	if err := cmd.MarkFlagRequired("image"); err != nil {
+		panic(err)
+	}
 
 	return cmd
 }
 
 func start(opts *StartOptions) error {
-	metrics.DefaultBindAddress = ":60000"
+	metrics.DefaultBindAddress = opts.MetricsListenAddr
 
 	// Get a kube client.
 	kubeConfig, err := config.GetConfig()
 	if err != nil {
-		log.Error(err, "failed to get kube config")
-		os.Exit(1)
+		return fmt.Errorf("failed to get kube config: %v", err)
 	}
 	kubeClient, err := operatorclient.NewClient(kubeConfig)
 	if err != nil {
-		log.Error(err, "failed to create kube client")
-		os.Exit(1)
+		return fmt.Errorf("failed to create kube client: %v", err)
 	}
 
-	// Collect operator configuration.
-	operatorNamespace := os.Getenv("WATCH_NAMESPACE")
-	if len(operatorNamespace) == 0 {
-		operatorNamespace = manifests.DefaultOperatorNamespace
-	}
-	log.Info("using operator namespace", "namespace", operatorNamespace)
+	log.Info("using operator namespace", "namespace", opts.OperatorNamespace)
 
-	ingressControllerImage := os.Getenv("IMAGE")
-	if len(ingressControllerImage) == 0 {
-		log.Error(fmt.Errorf("missing environment variable"), "'IMAGE' environment variable must be set")
-		os.Exit(1)
-	}
-	releaseVersion := os.Getenv("RELEASE_VERSION")
-	if len(releaseVersion) == 0 {
-		releaseVersion = statuscontroller.UnknownVersionValue
+	if opts.ReleaseVersion == statuscontroller.UnknownVersionValue {
 		log.Info("RELEASE_VERSION environment variable missing", "release version", statuscontroller.UnknownVersionValue)
```

Comment:
Update the log message not to refer to the environment variable.

Reply:
Fixed